api/src/main.rs

```diff
@@ -1,37 +1,217 @@
+                                return Outcome::Failure((
+                                    rocket::http::Status::Unauthorized,
+                                    UserError::TokenError,
+                                ))
+                            }
+                        };
+
+                        Outcome::Success(Self {
+                            email: typed_token.sub,
+                        })
+                    },
+                )
+            }
+            None => request::Outcome::Failure((
+                rocket::http::Status::Unauthorized,
+                UserError::MissingToken,
+            )),
+        }
+    }
+}
+
+impl<'a> OpenApiFromRequest<'a> for User {
+    fn from_request_input(
+        _gen: &mut OpenApiGenerator,
+        _name: String,
+        _required: bool,
+    ) -> rocket_okapi::Result<RequestHeaderInput> {
+        // Setup global requirement for Security scheme
+        let security_scheme = SecurityScheme {
+            description: Some(
+                "Requires an Bearer token to access, token is: `mytoken`.".to_owned(),
+            ),
+            // Setup data requirements.
+            // In this case the header `Authorization: mytoken` needs to be set.
+            data: SecuritySchemeData::Http {
+                scheme: "bearer".to_owned(), // `basic`, `digest`, ...
+                // Just gives use a hint to the format used
+                bearer_format: Some("bearer".to_owned()),
+            },
+            extensions: Object::default(),
+        };
+        // Add the requirement for this route/endpoint
+        // This can change between routes.
+        let mut security_req = SecurityRequirement::new();
+        // Each security requirement needs to be met before access is allowed.
+        security_req.insert("User".to_owned(), Vec::new());
+        // These vvvvvvv-----^^^^^^^^ values need to match exactly!
+        Ok(RequestHeaderInput::Security(
+            "User".to_owned(),
+            security_scheme,
+            security_req,
+        ))
+    }
+}
+
+#[derive(Serialize, JsonSchema)]
+#[serde(crate = "rocket::serde")]
+struct RootResponse {
+    value: String,
+}
+
+#[openapi]
+#[get("/", format = "json")]
+async fn index(
+    pool: &State<PgPool>,
+    user: User,
+) -> Result<Json<RootResponse>, rocket::response::status::BadRequest<String>> {
     // Make a simple query to return the given parameter (use a question mark `?` instead of `$1` for MySQL)
     let test_id: i32 = match sqlx::query!("SELECT id FROM test")
         .fetch_one(pool.inner())
         .await
     {
         Ok(test_id) => test_id.id,
-        Err(e) => {
-            // One day we should handle errors better (HTTP return codes)
-            println!("Error: {}", e);
-            return "Error".to_string();
+        Err(_) => {
+            return Err(rocket::response::status::BadRequest(Some(
+                "Error getting database ID".to_string(),
```

Comment:
the error might contain more information that is useful to the user? Instead of _ throwing it away, include it in the bad request?